attack.go

```diff
@@ -176,7 +176,8 @@ func attack(opts *attackOpts) (err error) {
 		vegeta.UnixSocket(opts.unixSocket),
 	)
 
-	res := atk.Attack(tr, opts.rate, opts.duration, opts.name)
+	rater := vegeta.NewFixedRater(opts.rate, opts.duration)
+	res := atk.Attack(tr, rater, opts.name)
```

Comment:
Drat. Preserving backwards compatibility isn't possible when pushing the responsibility for counting hits to the `Rater`, because the Rate struct needs to have a pointer receiver for its methods. That's why it's wrapped with the `FixedRater` struct here.

Reply:
I'm on-board with trying to minimize breaking backwards compatibility, despite Go modules and vendoring making this less of a problem than it was in the past.